### PR TITLE
Fix metadata loss across state-restore.

### PR DESCRIPTION
**Pull request by tokatoka**
state.add_metada() deletes past metadata

### CODE REVIEW

File: libafl/src/schedulers/accounting.rs [hunk 1]
```diff
@@ -263,7 +280,16 @@ where
     /// Creates a new [`CoverageAccountingScheduler`] that wraps a `base` [`Scheduler`]
     /// and has a default probability to skip non-faved [`Testcase`]s of [`DEFAULT_SKIP_NON_FAVORED_PROB`].
     pub fn new(state: &mut S, base: CS, accounting_map: &'a [u32]) -> Self {
-        state.add_metadata(TopAccountingMetadata::new(accounting_map.len()));
+        match state.metadata().get::<TopAccountingMetadata>() {
+            Some(meta) => {
+                if meta.max_accounting.len() != accounting_map.len() {
```
Comment (domenukk) [inline libafl/src/schedulers/accounting.rs:285]: This will still override the contents, is that fine?

File: libafl/src/schedulers/accounting.rs [hunk 2]
```diff
@@ -88,6 +88,23 @@ impl TopAccountingMetadata {
...
     }
+    /// Getter
+    #[must_use]
+    pub fn map(&self) -> &HashMap<usize, usize> {
+        &self.map
+    }
+
+    /// Getter
+    #[must_use]
+    pub fn changed(&self) -> bool {
```
Comment (domenukk) [inline libafl/src/schedulers/accounting.rs:99]: Why have a method when the value is already pub?

File: libafl/src/schedulers/accounting.rs (hunk 1 above)
Comment (andreafioraldi) [reply libafl/src/schedulers/accounting.rs:285]: it seems fine

**Comment by andreafioraldi** [comment]
@tokatoka if this PR is ready merge it ASAP

**Pull request merged**